### PR TITLE
ENH: add `nanmax`

### PR DESCRIPTION
**Pull request by OmarManzoor**
Towards #789

- Adds support for `nanmax`

### CODE REVIEW

**Comment by lucascolley** [comment]
as mentioned earlier, would be nice to see a branch with this and `nanmin` working in sklearn

**Comment by OmarManzoor** [comment]
How should we test? Will you merge this and then I need to adjust the version of `array-api-extra` in `scikit-learn` on my local system?

**Comment by lucascolley** [comment]
you can checkout this branch of array-api-extra inside sklearn

**Comment by OmarManzoor** [comment]
The CI in `scikit-learn` is 🟢

**Comment by OmarManzoor** [comment]
@lucascolley Can we merge this one?

File: tests/test_funcs.py [hunk 1]
```diff
@@ -2285,3 +2286,84 @@ def test_xp(self, axis: int | None, expected_list: list[float], xp: ModuleType):
         res = nanmin(a, axis=axis, xp=xp)
         expected = xp.asarray(expected_list)
         assert_equal(res, expected)
+
+
+class TestNanMax:
```
Comment (lucascolley) [inline tests/test_funcs.py:2291]: can we reduce the duplication with `TestNanMin` a bit?

**Review by lucascolley — CHANGES_REQUESTED** [review]

File: tests/test_funcs.py (hunk 1 above)
Comment (OmarManzoor) [reply tests/test_funcs.py:2291]: What do you propose?

Comment (lucascolley) [reply tests/test_funcs.py:2291]: hmm, I figured we could define test cases at the class level and parametrize the class with both functions, but maybe that is more effort than it is worth given the different data for each.

Comment (OmarManzoor) [reply tests/test_funcs.py:2291]: We could try to keep the input data the same but the answers will need to be parametrized as they will be different for both functions

Comment (lucascolley) [reply tests/test_funcs.py:2291]: If you want to try something then go for it, otherwise I won't block, I guess this is clean enough already :)

Comment (OmarManzoor) [reply tests/test_funcs.py:2291]: If I add parametrizations for each of tests it might get a bit awkward adding functions + axis + expected results in the parameters. In my quick overview I am not sure how that will look as compared to the current separation which doesn't have any long parametrizations.

Comment (lucascolley) [reply tests/test_funcs.py:2291]: okay yeah let's leave it as is

**Review by lucascolley — APPROVED** [review]
thanks Omar!
